master/buildbot/db/changes.py

```diff
@@ -195,7 +195,8 @@ def getChangesForBuild(self, buildid):
 
         # For each codebase, append changes until we match the parent
         for cb, change in iteritems(fromChanges):
-            if change and change['changeid'] != toChanges.get(cb, {}).get('changeid'):
+            # Careful; toChanges[cb] may be None from getChangeFromSSid
+            if change and change['changeid'] != (toChanges.get(cb) or {}).get('changeid'):
```

Comment:
Indeed.

    .get(cb) or {}

is not strictly equivalent to

    .get(cb, {})

As it might return None if [cb] is defined to None.
I am not very fond of this way to write it.

Can we calculate toChangeId before the loop, so that this is easier to read?